src/components/AppNavigationItem/AppNavigationItem.vue

```diff
@@ -19,93 +20,253 @@
-		<!-- Bullet -->
-		<div v-if="item.bullet" :style="{ backgroundColor: item.bullet }" class="app-navigation-entry-bullet" />
-
-		<button v-if="collapsible" class="collapse" @click.prevent.stop="toggleCollapse" />
-
-		<a :class="item.icon" :href="(item.href) ? item.href : '#'"
-			@click="callPreventStop(simpleAction, $event)">
-			<img v-if="item.iconUrl" :alt="item.text" :src="item.iconUrl">
-			{{ item.text }}
-		</a>
+<docs>
 
-		<!-- Popover, counter and button(s) -->
-		<div v-if="item.utils" class="app-navigation-entry-utils">
-			<ul>
-				<!-- counter -->
-				<AppNavigationCounter v-if="item.utils.counter" :highlighted="item.utils.counter_highlighted === true">
-					{{ item.utils.counter }}
-				</AppNavigationCounter>
-
-				<!-- first action if only one action -->
-				<li v-if="item.utils.actions && item.utils.actions.length === 1"
-					class="app-navigation-entry-utils-menu-button">
-					<button :class="item.utils.actions[0].icon" :title="item.utils.actions[0].text" @click="item.utils.actions[0].action" />
-				</li>
-
-				<!-- menu button if at least two actions -->
-				<li v-else-if="item.utils.actions && item.utils.actions.length > 1"
-					class="app-navigation-entry-utils-menu-button">
-					<button v-click-outside="hideMenu" @click="toggleMenu" />
-				</li>
-			</ul>
-		</div>
+# Usage
 
-		<!-- menu if at least two actions -->
-		<div v-if="item.utils && item.utils.actions && item.utils.actions.length > 1"
-			:class="{ 'open': openedMenu }" class="app-navigation-entry-menu">
-			<PopoverMenu :menu="item.utils.actions" />
-		</div>
+### Simple element
+
+* With an icon:
+
+```
+<AppNavigationItem title="My title" icon="icon-category-enabled">
+
+```
+* With a spinning loader instead of the icon:
+
+<AppNavigationItem title="Loading Item" :loading="true" />
+
+### Element with actions
+Wrap the children in a template. If you have more than 2 actions, a popover menu and a menu
+button will be automatically created.
+
+```
+<AppNavigationItem title="Item with actions" icon="icon-category-enabled">
+	<template slot="actions">
+		<ActionButton icon="icon-edit" @click="alert('Edit')">
+			Edit
+		</ActionButton>
+		<ActionButton icon="icon-delete" @click="alert('Delete')">
+			Delete
+		</ActionButton>
+		<ActionLink icon="icon-external" title="Link" href="https://nextcloud.com" />
+	</template>
+</AppNavigationItem>
+```
+
+### Element with counter
+Just nest the counter into <AppNavigationItem> and add `slot="counter"` to it.
+
+```
+<AppNavigationItem title="Item with counter" icon="icon-folder">
+	<AppNavigationCounter slot="counter">
+		99+
+	</AppNavigationCounter>
+</AppNavigationItem>
+```
+
+### Element with children
+
+Wrap the children in a template with the `slot` property and use the prop `allowCollapse` to choose wether to allow or
+prevent the user from collapsing the items.
+
+```
+<AppNavigationItem title="Item with children" :allowCollapse="true">
+	<template slot="children">
```

Comment:
This doesn't work. I think it must be `<template>`.

Reply:
> This doesn't work. I think it must be <template>.

I don't get it, what do you mean exactly??

Reply:
Right, we remove the scoping of the children scope!